Discover-for-Cloud-Templates/AWS/troubleshooting/scripts/falcon_discover_accounts.py

```diff
@@ -33,102 +31,124 @@ def format_api_payload(rate_limit_reqs=0, rate_limit_time=0):
                     }
-                    #Use the account_value function to retrieve the access_health branch, which contains our api failure reason
+                    # Use the account_value function to retrieve the
+                    # access_health branch, which contains our api failure reason
                     try:
-                        print(f'Account {result["id"]} has a problem: {account_value(result["id"],"access_health")["api"]["reason"]}')
-                    except:
-                        #The above call will produce an error if we're running check immediately after registering an account as
-                        #the access_health branch hasn't been populated yet. Requery the API for the account_list when this happens.
-                        account_list = falcon_discover.QueryAWSAccounts(parameters={"limit":f"{str(query_limit)}"})["body"]["resources"]
-                        print(f'Account {result["id"]} has a problem: {account_value(result["id"],"access_health")["api"]["reason"]}')
-                    #Output the account details to the user to assist with troubleshooting the account
+                        problem = account_value(result["id"], "access_health")["api"]["reason"]
+                        print(f'Account {result["id"]} has a problem: {problem}')
+                    except:  # noqa: E722
+                        # The above call will produce an error if we're running
+                        # check immediately after registering an account as
+                        # the access_health branch hasn't been populated yet.
+                        # Requery the API for the account_list when this happens.
+                        account_list = falcon.QueryAWSAccounts(
+                            parameters={"limit": "{}".format(str(query_limit))}
+                            )["body"]["resources"]
+                        issue = account_value(result["id"], "access_health")["api"]["reason"]
+                        print(f'Account {result["id"]} has a problem: {issue}')
+                    # Output the account details to the user to assist with troubleshooting the account
                     print(f'Current settings {json.dumps(account_values_to_check, indent=4)}\n')
+            for status in [status for status in (access_response["body"]["errors"] or [])]:
+                # This account has an issue
+                print(f'Account {status["id"]} has an error: {status["message"]}!')
         else:
             try:
-                #An error has occurred
-                print("Got response error code {} message {}".format(access_response["status_code"], access_response["body"]["errors"][0]["message"]))
-            except:
-                #Handle any egregious errors that break our return error payload
+                # An error has occurred
+                print("Got response error code {}".format(access_response["status_code"]))
+            except:  # noqa: E722
```

Comment:
QQ: Why not do at a minimum of `except Exception as e` ? Do we really not know the exception here?

Reply:
Could be one of a few variations depending.  Since this is just a reporting function that outputs to the screen on the status of the content, it didn't seem necessary to explicitly detail them all.